### DISCUSSION TITLE
Trigger default branch workflow on fast forward merge push.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#27216] [Actions]
<p>Hi! New to github actions and trying to set things up.</p>
<p>I spent a good amount of time in github actions docs but I have difficulties to express my problem. Therefore I may have missed the relevant part…</p>
<p>Anyway, I use the following workflow :</p>
<pre><code>$ git checkout master # here is my default branch
$ git checkout develop # here is my development/unstable branch
$ # make some changes
$ git add ...
$ git commit ...
$ # repeat while new feature development is finished ...
$ git push # optionally push to origin/develop branch -&gt; triggers workflow -&gt; passed -&gt; all OK
$ git checkout master
$ git merge --ff-only develop
$ git push # push to origin/master -&gt; triggers workflow on develop branch instead of master =(
</code></pre>
<p>With the following configuration (rust specific):</p>
<pre><code>name: Rust

on: [push]

jobs:
    build:

        runs-on: ubuntu-latest

        steps:
        - uses: actions/checkout@v2
        - name: Build
          run: cargo build --verbose
        - name: Run tests
          run: cargo test --verbose
</code></pre>
<p>Pushing the master branch with only a fast forward merge does not trigger a build on master branch but on the develop branch. <strong>Is it possible to trigger default branch build (here master) when pushing only a fast-forward merge to origin ?</strong></p>
<p>Thanks for your help.</p>
<ul>
<li>logan</li>
</ul>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Sorry for letting you suffer this issue. This is an known issue for fast-forward merge. Github Team is tracking on it. </p>
<p>Currently, I would not encourage you to use fast forward merge when push to master.  </p>

**Comment by ghost** [comment]
<p>Thanks for your answer!</p>

**Comment by Svalorzen** [comment]
<p>Hi! I’m also affected by this problem, and I have not found any other reference to it by googling aside from this thread.</p>
<p>Is there any URL I could bookmark that tracks this issue? Thanks!</p>
